lib/ansible/plugins/filter/ipaddr.py

```diff
@@ -767,6 +770,23 @@ def ipsubnet(value, query='', index='x'):
                     return str(value.supernet(query)[0])
                 except:
                     return False
+    elif querystr:
+        try:
+            vtype = ipaddr(query, 'type')
+            if vtype == 'address':
+                v = ipaddr(query, 'cidr')
+            elif vtype == 'network':
+                v = ipaddr(query, 'subnet')
+
+            query = netaddr.IPNetwork(v)
+        except:
+            return False
+
+        if network_in_network(query, value):
+            subnetlist = list(query.subnet(value.prefixlen))
```

Comment:
According to the doc, it's a generator: https://netaddr.readthedocs.io/en/latest/api.html#netaddr.IPNetwork.subnet

So you can iterate over it, no need to `list()` it. Please remove it.